pkg/controller/constrainttemplate/constrainttemplate_controller_test.go

```diff
@@ -873,6 +881,9 @@ func TestReconcile(t *testing.T) {
 			if vapBindingCreationTime.Before(blockTime) {
 				return fmt.Errorf("VAPBinding should not be created before the timestamp")
 			}
+			if ct.GetAnnotations()[constraint.VAPBGenerationAnnotation] == constraint.VAPBGenerationUnblocked {
+				return fmt.Errorf("expected %s annotations on CT to be unblocked", constraint.VAPBGenerationAnnotation)
+			}
```

Comment:
same here

Reply:
Updated the test

Reply:
Did the tests fail before you updated this test logic?

Reply:
No, but in that case the condition was not evaluating to true (due to CT reconciler being slightly slow to update the annotation than constraint controller in creating vapb once the "time window" is crossed) so the error was not being returned.

Reply:
Then does that mean this test will be flaky since the condition will be true when CT reconciler fails to update the annotation?

Reply:
No, because by the time of next retry CT reconciler would have updated the template and it should pass as expected.

Reply:
For completeness, looks like the failed test gets retried:
> err = retry.OnError(testutils.ConstantRetry, func(_ error) bool {